src/decisionengine/framework/dataspace/datasources/postgresql.py

```diff
@@ -122,12 +122,18 @@ class Postgresql(ds.DataSource):
 
     def __init__(self, config_dict):
         super().__init__(config_dict)
+        self.logger = logging.getLogger()
+        self.logger.debug('Initializing a Postgresql datasource')
+
+        try:
+            self.logger.debug(f"Trying to connect as {config_dict['user']}@{config_dict['host']}:{config_dict['port']}/{config_dict['database']}")
+        except KeyError:
+            self.logger.debug(f"Trying to connect as {config_dict['user']}@{config_dict['host']}:{config_dict['port']}/{config_dict['dbname']}")
```

Comment:
It looks to me that the potential `KeyError` applies to `config_dict['database']` vs. `config_dict['dbname']` calls, which are somewhat buried in the f-strings, making the code difficult to read.  Perhaps the following is better:

```python
dbname = config_dict.get('database', config_dict['dbname'])
self.logger.debug(f"Trying to connect as {config_dict['user']}@{config_dict['host']}:{config_dict['port']}/{dbname}")
```

or perhaps a separately named function:

```python
dbname = require_one_of(config_dict, 'database', 'dbname')
self.logger.debug(f"Trying to connect as {config_dict['user']}@{config_dict['host']}:{config_dict['port']}/{dbname}")
```

where `require_one_of` has the desired semantics.

Reply:
I've reworked this based on your feedback.